### PR TITLE
feat(roll): roll Playwright 1.17.0-next-1636463074000

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright-python/issues/1004

### CODE REVIEW

File: playwright/_impl/_fetch.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        request = urlOrRequest if isinstance(urlOrRequest, network.Request) else None
+        assert request or isinstance(
+            urlOrRequest, str
+        ), "First argument must be either URL string or Request"
+        assert (
+            len(
+                list(
+                    filter(
+                        lambda entry: entry is True,
+                        [data is not None, form is not None, multipart is not None],
```
Comment (pavelfeldman) [inline playwright/_impl/_fetch.py:247]: (1 if data else 0) + (1 if form else 0) + (1 if multipart else 0) <= 1

File: playwright/_impl/_fetch.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    def __init__(self, context: APIRequestContext, initializer: Dict) -> None:
+        self._loop = context._loop
+        self._dispatcher_fiber = context._connection._dispatcher_fiber
+        self._request = context
+        self._initializer = initializer
+        self._headers = network.RawHeaders(initializer["headers"])
+
+    @property
+    def ok(self) -> bool:
+        return self.status == 0 or (self.status >= 200 and self.status <= 299)
```
Comment (pavelfeldman) [inline playwright/_impl/_fetch.py:350]: Let's ask Yury why this is so.

File: playwright/_impl/_helper.py [hunk 3]
```diff
@@ -275,3 +275,21 @@ def to_impl(obj: T) -> T:
...
         return cast(Any, obj)._impl_obj
     return obj
+
+
+def object_to_array(obj: Optional[Dict]) -> Optional[List[NameValue]]:
+    if not obj:
+        return None
+    result = []
+    for key, value in obj.items():
+        result.append(NameValue(name=key, value=str(value)))
```
Comment (pavelfeldman) [inline playwright/_impl/_helper.py:285]: I'm sure we have such method for headers (same types)

File: playwright/_impl/_fetch.py [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+            if isinstance(data, str):
+                post_data_buffer = data.encode()
+            elif isinstance(data, bytes):
+                post_data_buffer = data
+            elif isinstance(data, (dict, list)):
+                json_data = data
+            else:
+                raise Error(f"Unsupported 'data' type: {type(data)}")
+        elif form:
+            form_data = object_to_array(form)
```
Comment (pavelfeldman) [inline playwright/_impl/_fetch.py:272]: object_to_string_array

File: playwright/_impl/_fetch.py [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+        elif multipart:
```
Comment (mxschmitt) [inline playwright/_impl/_fetch.py:273]: check if there are multi values for form/multipart are possible.

File: playwright/_impl/_fetch.py [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+        )
+
+    async def fetch(
+        self,
+        urlOrRequest: Union[str, network.Request],
+        params: Dict[str, Union[bool, float, str]] = None,
+        method: str = None,
+        headers: Headers = None,
+        data: Union[Any, bytes, str] = None,
+        form: Dict[str, Union[bool, float, str]] = None,
```
Comment (pavelfeldman) [inline playwright/_impl/_fetch.py:232]: Let's ask Yury about repeated keys.

**Review by pavelfeldman — APPROVED** [review]

File: playwright/_impl/_fetch.py (hunk 2 above)
Comment (yury-s) [reply playwright/_impl/_fetch.py:350]: copied from another place

File: playwright/_impl/_fetch.py (hunk 5 above)
Comment (yury-s) [reply playwright/_impl/_fetch.py:273]: we discussed this and decided to use a kv map for simplicity of the api, multivalue should be exotic

File: playwright/_impl/_fetch.py (hunk 6 above)
Comment (mxschmitt) [reply playwright/_impl/_fetch.py:232]: answered in https://github.com/microsoft/playwright-python/pull/1013#discussion_r745931594

File: playwright/_impl/_fetch.py (hunk 4 above)
Comment (mxschmitt) [reply playwright/_impl/_fetch.py:272]: headers have more edge-cases for e.g. set-cookies etc.

File: playwright/_impl/_helper.py (hunk 3 above)
Comment (mxschmitt) [reply playwright/_impl/_helper.py:285]: headers have more edge-cases for e.g. set-cookies etc.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**